MonkeyBot/Modules/Polls/PollService.cs

```diff
@@ -28,50 +31,77 @@ public async Task AddAndStartPollAsync(Poll poll)
             DiscordGuild guild = await _discordClient.GetGuildAsync(poll.GuildId);
             DiscordChannel channel = guild?.GetChannel(poll.ChannelId);
             DiscordMessage pollMessage = await channel.GetMessageAsync(poll.MessageId);
-            Dictionary<DiscordEmoji, string> emojiMapping = GetEmojiMapping(poll.PossibleAnswers);
-
-            foreach (DiscordEmoji pollEmoji in emojiMapping.Keys)
-            {
-                await pollMessage.CreateReactionAsync(pollEmoji);
-            }
+            
             await pollMessage.PinAsync();
 
             await _dbContext.Polls.AddAsync(poll);
             await _dbContext.SaveChangesAsync();
 
             await StartPollAsync(poll);
         }
-
+        
         private async Task StartPollAsync(Poll poll)
         {
             DiscordGuild guild = await _discordClient.GetGuildAsync(poll.GuildId);
             DiscordChannel channel = guild?.GetChannel(poll.ChannelId);
             DiscordMember pollCreator = await guild.GetMemberAsync(poll.CreatorId);
-            Dictionary<DiscordEmoji, string> emojiMapping = GetEmojiMapping(poll.PossibleAnswers);
-
-            TimeSpan pollDuration = poll.EndTimeUTC - DateTime.UtcNow;
-            await Task.Delay(pollDuration);
-            //Only the get the message once the end time is reached to get all reactions
             DiscordMessage pollMessage = await channel.GetMessageAsync(poll.MessageId);
-            var pollResult = pollMessage.Reactions
-                .Where(r => emojiMapping.Keys.Contains(r.Emoji))
-                .Select(r => (Emoji: r.Emoji, Count: r.Count - 1))
+            
+            pollMessage = await pollMessage.ModifyAsync(x => x.AddComponents(PollMessageUpdater.BuildAnswerButtons(poll.PossibleAnswers)));
```

Comment:
Not sure if we need a call to clear components first? Or does it silently substitute?

Reply:
For some reason channel.GetMessageAsync doesn't include components of message, therefore I didn't add ClearComponents.

Reply:
Fair enough 👍